src/argilla/client/feedback/training/base.py

```diff
@@ -33,7 +34,7 @@ class ArgillaTrainer(ArgillaTrainerV1):
     def __init__(
         self,
         dataset: "FeedbackDataset",
-        task_mapping: TrainingTaskMappingForTextClassification,
+        task: TrainingTaskTypes,
         framework: Framework,
         lang: Optional["spacy.Language"] = None,
         model: Optional[str] = None,
```

Comment:
This a cross-cutting issue, not coming from this PR, but it's specially relevant for LLMs:

Allowing users to pass only a `str` model ref is highly inflexible. Very often you want to configure your model with certain parameters during model instantiation (e.g., `AutoModelForCausalLM.from_pretrained`). Enabling users to pass a model (and tokenizer) object would result in a much more flexible API. 

Another potential solution would be to define a new arg like the (trainer) `framework_kwargs` but for model (and tokenizer). But I think this is less desirable in terms of developer experience and maintenance from our side.

Another alternative is to tell our users: if you want this level of flexibility just use `prepare_for_training` and do the training directly with trl, transformers, etc. But I think, if feasible, we should try to give this extra-level of flexibility even if ArgillaTrainer is a short-cut for less critical/customized training workflows.

A clear example of why users will need more flexibility is quantization (needed to train larger models with limited compute resources):

```python
compute_dtype = getattr(torch, bnb_4bit_compute_dtype)
bnb_config = BitsAndBytesConfig(
    load_in_4bit=use_4bit,
    bnb_4bit_quant_type=bnb_4bit_quant_type,
    bnb_4bit_compute_dtype=compute_dtype,
    bnb_4bit_use_double_quant=use_nested_quant,
)

model = AutoModelForCausalLM.from_pretrained(
    model_name,
    quantization_config=bnb_config,
    device_map=device_map
)

```

To the best of my knowledge, using this "configured" `model` is not possible with the current `ArgillaTrainer`. I'm aware this comes from the original ArgillaTrainer API, but maybe it's a good opportunity to tackle this now?

Love to hear your thoughts @davidberenstein1957 @tomaarsen @alvarobartt

Reply:
Any thoughts?

Reply:
I intended to review this with Alvaro and David today, but I didn't get around to it.

I think it should be feasible to allow either a model string or an initialized model. We would then also expose an optional tokenizer parameter. I think this should only require minimal changes, although I think it's best to create these changes in a separate PR.

Reply:
Great, I agree with creating a separate PR for this.